diff --git a/includes/Request/WebRequest.php b/includes/Request/WebRequest.php
index 734076449b8..78acae64100 100644
--- a/includes/Request/WebRequest.php
+++ b/includes/Request/WebRequest.php
@@ -288,10 +288,10 @@ class WebRequest {
 				$port = $stdPort;
 			} elseif ( $parts[1] === false ) {
 				if ( isset( $_SERVER['SERVER_PORT'] ) ) {
-					$port = $_SERVER['SERVER_PORT'];
+					$port = (int)$_SERVER['SERVER_PORT'];
 				} // else leave it as $stdPort
 			} else {
-				$port = $parts[1];
+				$port = (int)$parts[1];
 			}
 			break;
 		}
